### ISSUE TITLE
Clockwork request recursion

### ISSUE THREAD

**Issue by Basje** [itsgoingd/clockwork-firefox#3]
I've been experimenting with Clockwork Firefox (copy of `master`, downloaded today), and integrated the server components (with Packagist, installed today) in one of our projects. Using [an old tutorial](http://hotcashew.com/2013/12/integrating-chromes-clockwork-custom-web-app/) I managed to get the basics working, so all requests send the Clockwork headers correctly, and the `/__clockwork/{id}` route properly returns the stored data. Currently I'm just testing with `PhpDataSource`. When opening the Developer Console (F12) in Firefox Developer Edition v57, it will display some data, but it's also generating new requests constantly, effectively running in a never-ending loop, and filling the harddisk in the process. Using the latest Clockwork Chrome from the Chrome store (up-to-date Chrome and extension as of today), this problem is not present.

Our project uses the [League/Route](http://route.thephpleague.com/) package for routing all requests. Every single route sends the Clockwork headers, and every request stores the data at the assigned destination. 

What I think is happening is that Clockwork Firefox detects the headers of the request for data it is sending to `/__clockwork/{id}` and then sends a new request for that request, which then sends new headers, and so on. Not sure if that is the intended behaviour or not, but since Chrome is not doing it, and I kind of expected it to work the way it works in Chrome, I think it's a bug in the Firefox version of Clockwork. 

Ignoring Clockwork headers when requesting Clockwork information from the server might fix this, but I haven't had the time to look into this yet.

If you need any extra information, please let me know!

*Instant update:* Our server is configured to redirect to a path with a slash at the end if that's missing. So maybe the redirect is to blame. Still, the problem does not exist in Chrome.

**Comment by itsgoingd** [comment]
Hey, thanks for the extensive report!

I believe you might be correct in your conclusion. In the Laravel integration we have an option to ignore particular URIs and the clockwork metadata URIs are ignored by default. You might want to do this in your custom integration as well, even if the issue is solved, to save some storage and system resources since the metadata is useless in this case anyway.

Also in Firefox we use a different API to watch for requests since the one we use in Chrome is not yet implemented, this might explain why the Chrome version works for you. Btw. did you compare with the Chrome version form webstore or GitHub? Firefox version is based on the upcoming Chrome 2.0 version (master branch on GitHub) which has completely new metadata handling code.

I also have some other ideas, but I will need to take a closer look first. I'll update the issue as soon as I figure something out.

**Comment by Basje** [comment]
You are very welcome!

After some more testing I've verified that the problem is the presence of the `X-Clockwork-Id` header on the Clockwork metadata URIs. Whether or not the `X-Clockwork-Version` header is present doesn't seem to have any effect. 

> You might want to do this in your custom integration as well, even if the issue is solved, to save some storage and system resources since the metadata is useless in this case anyway.

Good point. As I was just testing, I used the easiest way to do this. Now that I've excluded the Clockwork metadata URIs from the Clockwork logic, the extension seems to be working as expected. So at least there is an easy way to mitigate the problem.
